### PR TITLE
Bryanv/build improvements

### PR DESCRIPTION
**Pull request by bryevdv**
This is a WIP PR to clean up some of our build code as well as provide some tools and docs to help new contributors get started more easily. 

The last commit relaxes some constraints in `meta.yaml` will probably have to experiment to get things right.

### CODE REVIEW

**Comment by bryevdv** [comment]
OK There is quite a bit more I want to do, but build work is tedious and exhausting so I think this is a good stopping place before embarking on adding Travis stages

File: .travis.yml [hunk 1]
```diff
@@ -33,11 +33,11 @@ matrix:
       env: GROUP=examples EXTRA_CMD=--no-js
 
     # linting, package and docs building (python 3.4)
-    - python: 3.4
+    - python: 3.6
       env: GROUP=flake_docs
 
     # integration tests (python 3.4)
```
Comment (mattpap) [inline .travis.yml]: Comments don't match. Actually I'm not sure what's the point of this repetition.

Comment (bryevdv) [inline .travis.yml]: Most of `.travis.yml` is going away in a follow on PR so not going to bother fixing at this point.

**Comment by bryevdv** [comment]
I'm going to go ahead and merge this, hopefully nothing to controversial in it. Mostly just some minor improvement/simplifications and better dev "getting started" docs. (Some dev docs were deleted I will put them back elsewhere later)

**Comment by mattpap** [comment]
It looks good, as much as I could review.

**Pull request merged**